src/frontend/mame/ui/utils.h

```diff
@@ -260,7 +260,7 @@ bool input_character(std::string &buffer, unicode_char unichar, F &&filter)
 			result = true;
 		}
 	}
-	else if ((unichar >= ' ') && filter(unichar))
+	else if ((unichar >= ' ') && (size == ~0 || buffer.size() < size) && filter(unichar))
```

Comment:
Try harder.  It's a faulty assumption that the old code was more complex than necessary.